src/core/server/logging/integration_tests/logging.test.ts

```diff
@@ -146,12 +146,18 @@ describe('logging service', () => {
       ],
     };
 
+    const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
     let root: ReturnType<typeof createRoot>;
     let setup: InternalCoreSetup;
     let mockConsoleLog: jest.SpyInstance;
     const loggingConfig$ = new Subject<LoggerContextConfigInput>();
-    const setContextConfig = (enable: boolean) =>
-      enable ? loggingConfig$.next(CUSTOM_LOGGING_CONFIG) : loggingConfig$.next({});
+    const setContextConfig = async (enable: boolean) => {
+      loggingConfig$.next(enable ? CUSTOM_LOGGING_CONFIG : {});
+      // need to wait for config to reload. nextTick is enough, using delay just to be sure
+      await delay(10);
```

Comment:
This is required because `LoggingService.configure` underlying call to `setContextConfig` (that is now asynchronous) is not awaited.

In real world scenario this is (probably) fine, as configure takes an observable as parameter, but in the tests, we were assuming the provided observable was immediately and synchronously triggering the config reload.

We could change the `LoggingService.setup.configure` implementation to return an observable that would emit when `loggingSystem.setContextConfig` resolves, but that's some additional complexity and didn't felt necessary.